internal/server/server_test.go

```diff
@@ -0,0 +1,67 @@
+package server
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/SOMAS2020/SOMAS2020/internal/common"
+	"github.com/SOMAS2020/SOMAS2020/pkg/testutils"
+)
+
+// a factory function that allows us to inject custom stuff to facilitate testing
+func somasServerTester(clients []common.Client, gameState common.GameState) Server {
+	return &SOMASServer{
+		clients:   clients,
+		gameState: gameState,
+	}
+}
+
+// extend as required
+type mockClient struct {

```

Comment:
this is somewhat unsafe, if some test code calls a function we didn't "override" here (we "overrode" `Echo` and `GetID` below), then a segfault will happen. This is test code, so if we safely use it, it should be fine. It is however discouraged, but implementing a complete client is kinda cumbersome--use at your own risk

Reply:
(`common.Client` is an interface -- I'm forcing mockClient to be an instance of common.Client here)